shared/AppInsightsCore/src/JavaScriptSDK/TelemetryPluginProxy.ts

```diff
@@ -0,0 +1,114 @@
+        let _self = this;
+        let _nextPlugin:ITelemetryPluginProxy = null;
+        let _hasRun = false;
+        let _hasProcessTelemetry = _isFunction(plugin.processTelemetry);
+        let _hasSetNext = _isFunction(plugin.setNextPlugin);
+
+        _self.hasRun = () => {
+            return _hasRun;
+        }
+
+        _self.getPlugin = () => {
+            return plugin;
+        };
+
+        _self.getNextPlugin = () => {
+            return _nextPlugin;
+        };
+
+        _self.setNextPlugin = (nextPlugin:ITelemetryPluginProxy) => {
+            _nextPlugin = nextPlugin;
+        }
+
+        _self.processTelemetry = (env: ITelemetryItem, itemCtx:ITelemetryItemContext) => {
+            if (!itemCtx) {
+                // Looks like a plugin didn't pass the (optional) context, so restore to the default
+                itemCtx = defItemCtx;
+            }
+
+            if (!_hasRun && plugin && _hasProcessTelemetry) {
+                _hasRun = true;
+                try {
+
+                    // Ensure that we keep the context in sync (for processNext()), just in case a plugin
+                    // doesn't calls processTelemetry() instead of itemContext.processNext() or some 
+                    // other form of error occurred
+                    itemCtx.setNext(_nextPlugin);
+                    if (_hasSetNext) {
+                        // Backward compatibility setting the next plugin on the instance
+                        plugin.setNextPlugin(_nextPlugin);
+                    }
+
+                    plugin.processTelemetry(env, itemCtx);
+                } catch (error) {
+                    if (!_nextPlugin || !_nextPlugin.hasRun()) {
+                        // Either we have no next plugin or the current one did not attempt to call the next plugin
+                        // Which means the current one is the root of the failure so log/report this failure
+                        itemCtx.log().throwInternal(
+                            LoggingSeverity.CRITICAL,
+                            _InternalMessageId.PluginException,
+                            "Plugin [" + plugin.identifier + "] failed during processTelemetry - " + error);
+                    }
+
+                    if (_nextPlugin && !_nextPlugin.hasRun()) {
+                        // As part of the failure the current plugin did not attempt to call the next plugin in the cahin
+                        // So rather than leave the pipeline dead in the water we call the next plugin
+                        _nextPlugin.processTelemetry(env, itemCtx);
+                    }
+                }
+            } else if (_nextPlugin) {
+                _hasRun = true;
+
+                // The underlying plugin is either not defined or does not have a processTelemetry implementation
+                // so we still want the next plugin to be executed.
+                _nextPlugin.processTelemetry(env, itemCtx);
```

Comment:
This case actually fixes an existing issue where if you added a new IPlugin which didn't implement the processTelemetry() between (based on the priority) 2 other ITelemetryPlugin instances the trailing chain would not get called and it would actually caused an exception.